isis/src/base/objs/UserInterface/UserInterface.cpp

```diff
@@ -408,13 +408,41 @@ namespace Isis {
 /**
  * This is used to display the updated parameters when an app is called with -last
  * 
- * @return QString returnString- the updated command line 
+ * @return QString currentCommandLine - the current command line stored from the last run of an app
  */
-QString UserInterface::buildNewCommandLine(){
-  QString returnString = "";
+QString UserInterface::buildNewCommandLine(QString currentCommandLine){
+  QString returnString = "";  
+  vector<QString> valuesToUpdate;
+
+  // loop user input cmd line & get updated params
   for(auto i:p_cmdline){
-    returnString += i;
-    returnString += " ";
+    QString inputParam = i;
+
+    // check for updated args in cmd line
+    if(inputParam.contains("=", Qt::CaseInsensitive)){
+      valuesToUpdate.push_back(inputParam);
+      }
+    }
+
+  // loop currentCommandLine and update returnString 
+  for(auto i:currentCommandLine.split(" ")){
+    QString paramVale = i;
+    
+    // loop values to update
+    for(auto j:valuesToUpdate){
+      QString updateVal = j;
+
+      // check to see if if the histoy param is the same as our input param
+      if(paramVale.split("=")[0].startsWith(updateVal.split("=")[0].toUpper())
```

Comment:
I'm concerned about the parsing in here (split, toUpper). Once the data struct that holds user parameters has been updated (IsisAml), we should just ask the Aml for the parameters and then stuff them into a string to return.  What about params that were defaulted in the previous one (i.e., not on the command line), but present in the new run?